src/server/debugcmd.cc

```diff
@@ -249,55 +250,98 @@ void DebugCmd::Load(string_view filename) {
   (*cntx_)->SendOk();
 }
 
-void DebugCmd::Populate(CmdArgList args) {
-  if (args.size() < 2 || args.size() > 5) {
-    return (*cntx_)->SendError(UnknownSubCmd("populate", "DEBUG"));
+optional<DebugCmd::PopulateOptions> DebugCmd::ParsePopulateArgs(CmdArgList args) {
+  if (args.size() < 2 || args.size() > 8) {
+    (*cntx_)->SendError(UnknownSubCmd("populate", "DEBUG"));
+    return nullopt;
   }
 
-  uint64_t total_count = 0;
-  if (!absl::SimpleAtoi(ArgS(args, 1), &total_count))
-    return (*cntx_)->SendError(kUintErr);
-  std::string_view prefix{"key"};
+  PopulateOptions options;
+  if (!absl::SimpleAtoi(ArgS(args, 1), &options.total_count)) {
+    (*cntx_)->SendError(kUintErr);
+    return nullopt;
+  }
 
   if (args.size() > 2) {
-    prefix = ArgS(args, 2);
+    options.prefix = ArgS(args, 2);
   }
-  uint32_t val_size = 0;
+
   if (args.size() > 3) {
-    std::string_view str = ArgS(args, 3);
-    if (!absl::SimpleAtoi(str, &val_size))
-      return (*cntx_)->SendError(kUintErr);
+    if (!absl::SimpleAtoi(ArgS(args, 3), &options.val_size)) {
+      (*cntx_)->SendError(kUintErr);
+      return nullopt;
+    }
   }
 
-  bool populate_random_values = false;
-  if (args.size() > 4) {
-    ToUpper(&args[4]);
-    std::string_view str = ArgS(args, 4);
-    if (str != "RAND") {
-      return (*cntx_)->SendError(kSyntaxErr);
+  size_t index = 4;
+  while (args.size() > index) {
```

Comment:
Any reason not to use a `for` loop here?

Reply:
fixing